tracing-subscriber/src/fmt/writer.rs

```diff
@@ -1,24 +1,94 @@
 //! Abstractions for creating [`io::Write`] instances.
 //!
 //! [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
-
-use std::io;
+use std::{
+    io,
+    sync::{Mutex, MutexGuard},
+};
 
 /// A type that can create [`io::Write`] instances.
 ///
-/// `MakeWriter` is used by [`fmt::Subscriber`] or [`fmt::Layer`] to print formatted text representations of
-/// [`Event`]s.
+/// `MakeWriter` is used by [`fmt::Subscriber`] or [`fmt::Layer`] to print
+/// formatted text representations of [`Event`]s.
+///
+/// This trait is already implemented for function pointers and
+/// immutably-borrowing closures that return an instance of [`io::Write`], such
+/// as [`io::stdout`] and [`io::stderr`]. Additionally, it is implemented for
+/// [`std::sync::Mutex`][mutex] when the tyoe inside the mutex implements
+/// [`io::Write`].
+///
+/// # Examples
+///
+/// The simplest usage is to pass in a named function that returns a writer. For
+/// example, to log all events to stderr, we could write:
+/// ```
+/// let subscriber = tracing_subscriber::fmt()
+///     .with_writer(std::io::stderr)
+///     .finish();
+/// # drop(subscriber);
+/// ```
+///
+/// Any function that returns a writer can be used:
+///
+/// ```
+/// fn make_my_great_writer() -> impl std::io::Write {
+///     // ...
+///     # std::io::stdout()
+/// }
+///
+/// let subscriber = tracing_subscriber::fmt()
+///     .with_writer(make_my_great_writer)
+///     .finish();
+/// # drop(subscriber);
+/// ```
+///
+/// A closure can be used to introduce arbitrary logic into how the writer is
+/// created. For example, this will send every 5th event to stderr, and all
+/// other events to stdout (why you would actually want to do this, I have no
+/// idea, but you _can_):
```

Comment:
Let's maybe remove the first-person perspective here? Something along the lines of "Consider this silly example: send every 5th event..."